cstrml/swork/src/tests.rs

```diff
@@ -1348,102 +1182,872 @@ fn ab_upgrade_should_failed_with_files_size_unmatch() {
+                error: 10,
+                message: Some("IdentityNotExist"),
+            });
+
+            register_identity(&alice, &a_pk, &a_pk);
+
+            add_wr(&a_pk, &WorkReport {
+                report_slot: 0,
+                used: 2000,
+                free: 0,
+                reported_files_size: 2,
+                reported_srd_root: hex::decode("00").unwrap(),
+                reported_files_root: hex::decode("11").unwrap()
+            });
+            add_wr(&b_pk, &WorkReport {
+                report_slot: 0,
+                used: 100,
+                free: 0,
+                reported_files_size: 2,
+                reported_srd_root: hex::decode("00").unwrap(),
+                reported_files_root: hex::decode("11").unwrap()
+            });
+
+            // alice is not the owner of the group
+            assert_noop!(Swork::join_group(
+                Origin::signed(bob.clone()),
+                alice.clone()
+            ),
+            DispatchError::Module {
+                index: 0,
+                error: 12,
+                message: Some("NotOwner"),
+            });
+
+            // Alice create a group and be the owner
+            assert_ok!(Swork::join_group(
+                Origin::signed(alice.clone()),
+                alice.clone()
+            ));
+
+            assert_eq!(Swork::identities(&alice).unwrap_or_default(), Identity {
+                anchor: a_pk.clone(),
+                group: Some(alice.clone())
+            });
+
+            // bob's used is 100
+            assert_noop!(Swork::join_group(
+                Origin::signed(bob.clone()),
+                alice.clone()
+            ),
+            DispatchError::Module {
+                index: 0,
+                error: 13,
+                message: Some("IllegalUsed"),
+            });
+
+            add_wr(&b_pk, &WorkReport {
+                report_slot: 0,
+                used: 0,
+                free: 0,
+                reported_files_size: 2,
+                reported_srd_root: hex::decode("00").unwrap(),
+                reported_files_root: hex::decode("11").unwrap()
+            });
+
+            // Bob join the alice's group
+            assert_ok!(Swork::join_group(
+                Origin::signed(bob.clone()),
+                alice.clone()
+            ));
+
+            assert_eq!(Swork::identities(&bob).unwrap_or_default(), Identity {
+                anchor: b_pk.clone(),
+                group: Some(alice.clone())
+            });
+
+            // bob already joined a group
+            assert_noop!(Swork::join_group(
+                Origin::signed(bob.clone()),
+                alice.clone()
+            ),
+            DispatchError::Module {
+                index: 0,
+                error: 11,
+                message: Some("AlreadyJoint"),
+            });
         });
 }
 
 #[test]
-fn bonds_limit_during_upgrade_should_work() {
+fn join_group_should_work_for_used_in_work_report() {
```

Comment:
This can be talked offline.

Reply:
Scenarios😂 :
![s1](https://user-images.githubusercontent.com/10335300/102953757-007cc380-450d-11eb-8abb-c975508c808f.jpeg)
![s2](https://user-images.githubusercontent.com/10335300/102953766-05417780-450d-11eb-9e8a-0c563a054b37.jpeg)
![s3](https://user-images.githubusercontent.com/10335300/102953767-05da0e00-450d-11eb-8045-3397c604152d.jpeg)
![s4](https://user-images.githubusercontent.com/10335300/102953769-0672a480-450d-11eb-9248-92680c049606.jpeg)